Participant.java

```diff
@@ -0,0 +1,9 @@
+public interface Participant {
```

Comment:
Would be better to make `Participant` a class. There is no need for an interface.